src/helm/benchmark/scenarios/grammar.py

```diff
@@ -0,0 +1,148 @@
+from collections import defaultdict
+from dataclasses import dataclass, field, replace
+from typing import List, Optional
+from helm.common.hierarchical_logger import hlog
+
+import dacite
+import re
+import yaml
+
+
+ROOT_CATEGORY = "Root"
+
+
+@dataclass
+class Expansion:
+    """
+    Example: text = "this is a ${Topic}"
+    """
+
+    text: str
+    tags: List[str] = field(default_factory=list)
+
+    def __post_init__(self):
+        self.children = re.split(r"(\${\w+})", self.text)
```

Comment:
I'm confused about what this does; could you provide an example?

I'm expecting this to be a `re.match` rather than a `re.split` e.g. for `"this is a ${Topic}"` a match would get `"${Topic}"`, but the split would get `["this is a ", ""]`... am I missing something?

Reply:
Example added.  It returns `['this is a ', '${Topic}', '']`